.github/workflows/test-workflow.yml

```diff
@@ -20,6 +20,14 @@ jobs:
       with:
         node-version: "${{ steps.nvm.outputs.NVMRC }}"
 
+    - name: Cache Dependencies
+      uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
```

Comment:
```suggestion
```
From the docs, it seems like this acts as a fallback -- we don't want this as node_modules would then not match the lockfile?

Reply:
@MattIPv4 I think we need this because this act as the key to fetch the cache! And it should be like

```diff
- ${{ runner.os }}-node-
+ ${{ runner.os }}-node-${{ steps.nvm.outputs.NVMRC }}-
```

Reply:
As we're caching `~/.npm`, my understanding is that it'd just hit the registry if the right data was missing from the cached data, so I think this 'lax restoration should be okay.

Reply:
@yg `${{ runner.os }}-node-` is less strict, but you could add in `${{ runner.os }}-node-${{ steps.nvm.outputs.NVMRC }}-` as a new item in the list, before `${{ runner.os }}-node-`.